OrbitLinuxTracing/UtilsTest.cpp

```diff
@@ -0,0 +1,127 @@
+#include <gmock/gmock-matchers.h>
+#include <gtest/gtest.h>
+#include <sys/syscall.h>
+
+#include <condition_variable>
+#include <mutex>
+#include <thread>
+
+#include "Utils.h"
+
+namespace LinuxTracing {
+
+TEST(ReadFile, ProcPidCommOfOrbitLinuxTracingTests) {
+  std::string filename = absl::StrFormat("/proc/%d/comm", getpid());
+  std::optional<std::string> returned_comm = ReadFile(filename);
+  // Comm values have a size limit of 15 characters.
+  std::string expected_comm =
+      std::string{"OrbitLinuxTracingTests"}.substr(0, 15).append("\n");
+  ASSERT_TRUE(returned_comm.has_value());
+  EXPECT_EQ(returned_comm.value(), expected_comm);
+}
+
+TEST(ExecuteCommand, EchoHelloWorld) {
+  std::string string_to_echo = "Hello, World!";
+  std::optional<std::string> returned_result =
+      ExecuteCommand(absl::StrFormat("echo %s", string_to_echo));
+  std::string expected_result = string_to_echo + "\n";
+  ASSERT_TRUE(returned_result.has_value());
+  EXPECT_EQ(returned_result.value(), expected_result);
+}
+
+TEST(ListThreads, OrbitLinuxTracingTestsMainAndAnother) {
+  pid_t main_tid = syscall(SYS_gettid);
+  pid_t thread_tid = -1;
+  std::vector<pid_t> returned_tids{};
+  std::mutex mutex;
+  std::condition_variable cv_tid;
+  std::condition_variable cv_list;
+
+  std::thread thread{[&] {
+    {
+      std::unique_lock<std::mutex> lock(mutex);
+      thread_tid = syscall(SYS_gettid);
+      cv_tid.notify_one();
+    }
+    {
+      std::unique_lock<std::mutex> lock(mutex);
+      while (returned_tids.empty()) {
+        cv_list.wait(lock);
+      }
+    }
+  }};
+
+  {
+    std::unique_lock<std::mutex> lock(mutex);
+    returned_tids = ListThreads(getpid());
+    cv_list.notify_one();
+  }
+  {
+    std::unique_lock<std::mutex> lock(mutex);
+    while (thread_tid == -1) {
+      cv_tid.wait(lock);
+    }
```

Comment:
I do not think you need this.. I suspect it is guaranteed that we have thread_tid initialised after return from join().

You will need to release the mutex before joining though.. otherwise it will deadlock, if I am reading this correctly.

Reply:
What if the threads are listed before the thread is started?

unique_lock is a RAII, it releases the mutex.

Reply:
Ok, that looks good.